### PR TITLE
#6605 Tool to find logical flow routes between apps

### PR DESCRIPTION
**Pull request by kamransaleem**

### CODE REVIEW

**Comment by kamransaleem** [comment]
had to download jgrapht version due to the following error 

```
java: cannot access org.jgrapht.Graph
  bad class file: /C:/Users/User/.m2/repository/org/jgrapht/jgrapht-core/1.5.1/jgrapht-core-1.5.1.jar!/org/jgrapht/Graph.class
    class file has wrong version 55.0, should be 52.0
    Please remove or make sure it appears in the correct subdirectory of the classpath.
```

File: waltz-data/src/main/java/org/finos/waltz/data/logical_flow/LogicalFlowDao.java [hunk 1]
```diff
@@ -75,6 +75,16 @@ public class LogicalFlowDao {
             LOGICAL_FLOW.TARGET_ENTITY_KIND,
             newArrayList(EntityKind.APPLICATION, EntityKind.ACTOR));
 
+    private static final Field<String> SOURCE_EXTERNAL_ID_FIELD = InlineSelectFieldFactory.mkExternalIdField(
+            LOGICAL_FLOW.SOURCE_ENTITY_ID,
+            LOGICAL_FLOW.SOURCE_ENTITY_KIND,
+            newArrayList(EntityKind.APPLICATION, EntityKind.ACTOR));
+
+    private static final Field<String> TARGET_EXTERNAL_ID_FIELD = InlineSelectFieldFactory.mkExternalIdField(
+                    LOGICAL_FLOW.TARGET_ENTITY_ID,
```
Comment (davidwatkins73) [inline waltz-data/src/main/java/org/finos/waltz/data/logical_flow/LogicalFlowDao.java:84]: nitpick: indentation inconsistent with blocks above

File: waltz-jobs/src/main/java/org/finos/waltz/jobs/tools/lineage/LineageDiscoveryReport.java [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+            return;
+        }
+
+        // Load reference data
+        List<Application> allApplications = applicationDao.findAll();
+        List<LogicalFlow> allActiveFlows = logicalFlowDao.findAllActive();
+        List<DataTypeDecorator> allDecorators = logicalFlowDecoratorDao.findAll();
+
+        List<RouteLookup> routeLookups = FindRoutes(logicalFlowDecoratorDao, allApplications, allActiveFlows, allDecorators, sourceTargetCodes);
+//        List<List<Object>> rows = prepareReportRows(routeLookups);
```
Comment (davidwatkins73) [inline waltz-jobs/src/main/java/org/finos/waltz/jobs/tools/lineage/LineageDiscoveryReport.java:101]: still lots of commented out code in this class...

File: waltz-data/src/main/java/org/finos/waltz/data/logical_flow/LogicalFlowDao.java [hunk 3]
```diff
@@ -413,6 +425,7 @@ private SelectJoinStep<Record> baseQuery() {
         return dsl
                 .select(LOGICAL_FLOW.fields())
                 .select(SOURCE_NAME_FIELD, TARGET_NAME_FIELD)
+                .select(SOURCE_EXTERNAL_ID_FIELD, TARGET_EXTERNAL_ID_FIELD)
```
Comment (davidwatkins73) [inline waltz-data/src/main/java/org/finos/waltz/data/logical_flow/LogicalFlowDao.java:428]: We'll need to check this in prod to make sure it doesn't degrade the qry performance too much

Comment (kamransaleem) [reply waltz-data/src/main/java/org/finos/waltz/data/logical_flow/LogicalFlowDao.java:428]: Agreed

**Comment by davidwatkins73** [comment]
#6605

**Pull request merged**